packages/cdk/stacks/StatelessResourcesStack.ts

```diff
@@ -0,0 +1,204 @@
+import {
+  StackProps,
+  Stack,
+  App,
+  Fn,
+  CfnOutput
+} from "aws-cdk-lib"
+import {
+  AccessLevel,
+  AllowedMethods,
+  FunctionEventType,
+  OriginRequestCookieBehavior,
+  OriginRequestHeaderBehavior,
+  OriginRequestPolicy,
+  ViewerProtocolPolicy
+} from "aws-cdk-lib/aws-cloudfront"
+import {RestApiOrigin, S3BucketOrigin} from "aws-cdk-lib/aws-cloudfront-origins"
+import {Bucket} from "aws-cdk-lib/aws-s3"
+
+import {RestApiGateway} from "../resources/RestApiGateway"
+import {CloudfrontFunction} from "../resources/Cloudfront/CloudfrontFunction"
+import {CloudfrontDistribution} from "../resources/CloudfrontDistribution"
+import {nagSuppressions} from "../nagSuppressions"
+
+export interface StatelessResourcesStackProps extends StackProps {
+  readonly serviceName: string
+  readonly stackName: string
+  readonly version: string
+}
+
+/**
+ * Clinical Prescription Tracker UI Stateful Resources
+
+ */
+
+export class StatelessResourcesStack extends Stack {
+  public constructor(scope: App, id: string, props: StatelessResourcesStackProps){
+    super(scope, id, props)
+
+    // Context
+    /* context values passed as --context cli arguments are passed as strings so coerce them to expected types*/
+
+    // Imports
+    const staticContentBucket = Bucket.fromBucketArn(
+      this, "StaticContentBucket", Fn.importValue(`${props.serviceName}-stateful-resources:StaticContentBucket:Arn`))
+
+    // Resources
+    // - API Gateway
+    const apiGateway = new RestApiGateway(this, "ApiGateway", {
+      serviceName: props.serviceName,
+      stackName: props.stackName
+    })
+
+    // --- Methods & Resources
+
+    // - Cloudfront
+    // --- Origins
+    const staticContentBucketOrigin = S3BucketOrigin.withOriginAccessControl(
+      staticContentBucket,
+      {
+        originAccessLevels: [AccessLevel.READ]
+      }
+    )
+
+    const apiGatewayOrigin = new RestApiOrigin(apiGateway.restApiGateway, {
+      customHeaders: {
+        "destination-apigw-id": apiGateway.restApiGateway.restApiId // for later apigw waf stuff
+      }
+    })
+
+    // --- Origin Request Policies
+    /* Allow all for now, may want to review these at a later stage */
+    const apiGatewayRequestPolicy = new OriginRequestPolicy(this, "apiGatewayRequestPolicy", {
+      cookieBehavior: OriginRequestCookieBehavior.all(),
+      headerBehavior: OriginRequestHeaderBehavior.all()
```

Comment:
this needs a originRequestPolicyName with stack name in it to prevent duplication